### PR TITLE
feat(python): AssumeRole support for AWS Credential Provider

### PR DESCRIPTION
**Pull request by nameexhaustion**
Adds support for using an assumed role to `CredentialProviderAWS`

```python
pl.scan_parquet(
    "s3://.../...",
    credential_provider=pl.CredentialProviderAWS(
        assume_role={
            "RoleArn": "arn:aws:iam::0000000:role/ExampleRole",
            "RoleSessionName": "example",
        }
    ),
)
```

ref https://github.com/pola-rs/polars/issues/18979
ref https://github.com/pola-rs/polars/issues/15838

### CODE REVIEW

File: py-polars/polars/io/cloud/_utils.py [hunk 1]
```diff
@@ -26,30 +26,54 @@ def _infer_cloud_type(
...
     return None
+
+
+def _get_path_scheme(path: str | Path) -> str | None:
+    splitted = str(path).split("://", maxsplit=1)
+
+    return None if not splitted else splitted[0]
+
+
+def _is_aws_cloud(scheme: str) -> bool:
```
Comment (nameexhaustion) [inline py-polars/polars/io/cloud/_utils.py:62]: drive-by - this lets us avoid an exhaustive match

File: py-polars/polars/io/cloud/credential_provider.py [hunk 2]
```diff
@@ -134,9 +183,10 @@ def __call__(self) -> CredentialProviderFunctionReturn:
 
         return {"bearer_token": self.creds.token}, (
             int(
-                expiry.replace(
-                    # Google auth does not set this properly
-                    tzinfo=zoneinfo.ZoneInfo("UTC")
+                (
+                    expiry.replace(tzinfo=zoneinfo.ZoneInfo("UTC"))
+                    if expiry.tzinfo is None
```
Comment (nameexhaustion) [inline py-polars/polars/io/cloud/credential_provider.py:188]: drive-by - in case a future version of google auth adds timezone info

**Review by ritchie46 — APPROVED** [review]
